effekt/jvm/src/main/scala/effekt/Driver.scala

```diff
@@ -108,6 +108,97 @@ trait Driver extends kiama.util.Compiler[EffektConfig, EffektError] { outer =>
     afterCompilation(source, config)(context)
   }
 
+  def generateCrashReport(e: Throwable, context: Context, config: EffektConfig): Unit = {
+    import java.nio.file.{Files, Paths}
+    import java.util.Properties
+
+    // Capture general information
+    val effektVersion = effekt.util.Version.effektVersion
+    val osInfo = System.getProperty("os.name") +
+      " (" + System.getProperty("os.arch") + ") " +
+      System.getProperty("os.version")
+
+    val jvmInfo = System.getProperty("java.version") +
+      " (" + System.getProperty("java.vendor") + ")"
+
+    val stackTrace = e.getStackTrace.map(line => "  at " + line.toString).mkString("\n")
+
+    // Collect backend details (if applicable)
+    val backendInfoName = config.backend() match {
+      case Backend(name, compiler, runner) => name
+      case null => s"Backend-specific info Name not retrievable"
+    }
+    val backendInfoCompiler = config.backend() match {
+      case Backend(name, compiler, runner) => compiler
+      case null => s"Backend-specific info Compiler not retrievable"
+    }
+    val backendInfoRunner = config.backend() match {
+      case Backend(name, compiler, runner) => runner
+      case null => s"Backend-specific info Runner not retrievable"
+    }
+
+    // Construct the report
+    val report =
+      s"""|=== Effekt Compiler Crash Report ===
+          |
+          |Where to report: https://github.com/effekt-lang/effekt/issues
+          |
+          |
+          |Effekt Compiler Version: $effektVersion
+          |
+          |Backend-specific info:
+          |  - Name: $backendInfoName
+          |  - Compiler: $backendInfoCompiler
+          |  - Runner: $backendInfoRunner
+          |
+          |Operating System: $osInfo
+          |
+          |JVM Version: $jvmInfo
+          |
+          |Full Stack Trace:
+          |$stackTrace
+          |""".stripMargin
+
+    val mdReport =
+      s"""|# === Effekt Compiler Crash Report ===
+          |
+          |**Where to report**: [https://github.com/effekt-lang/effekt/issues](https://github.com/effekt-lang/effekt/issues)
+          |
+          |---
+          |## Compiler Details
+          |
+          |- **Effekt Compiler Version**: $effektVersion
+          |- **Backend-specific info**:
+          |  - **Name**: $backendInfoName
+          |  - **Compiler**: $backendInfoCompiler
+          |  - **Runner**: $backendInfoRunner
+          |
+          |- **Operating System**: $osInfo
+          |
+          |- **JVM Version**: $jvmInfo
+          |
+          |
+          |# Full Stack Trace:
+          |```
+          |$stackTrace
+          |```
+          |""".stripMargin
+
+    // Print the report to console
+    context.info(report)
+
+    // write to a markdown file in ./out/
+    try {
+      val outputPath = Paths.get("./out/crash-report.md")
```

Comment:
Is `out/` the correct folder here? What if the user overrode it in the config? :)
See how we deal with this in the Runner:
https://github.com/effekt-lang/effekt/blob/ff669930823f9311ce2e0df387f6c33a5c62d971/effekt/jvm/src/main/scala/effekt/Runner.scala#L165

Reply:
Great Idea,thank you!:)